File: api/src/main/java/com/brocodeactivist/recipient/services/RecipientService.java
package com.brocodeactivist.recipient.services;

import com.brocodeactivist.recipient.model.Recipient;

import java.util.List;

public interface RecipientService {
    Recipient createRecipient(Recipient recipient);

    List<Recipient> getAllRecipients();

    boolean deleteRecipient(Long id);

    Recipient getRecipientById(Long id);

    Recipient updateRecipient(Long id, Recipient recipient);
}

File: ui/src/services/RecipientService.js
import axios from "axios";

const RECIPIENT_API_BASE_URL = "http://localhost:8080/api/v1/recipients";

class RecipientService {
  saveRecipient(recipient) {
    return axios.post(RECIPIENT_API_BASE_URL, recipient);
  }

  getRecipients() {
    return axios.get(RECIPIENT_API_BASE_URL);
  }

  deleteRecipient(id) {
    return axios.delete(RECIPIENT_API_BASE_URL + "/" + id);
  }

  getRecipientById(id) {
    return axios.get(RECIPIENT_API_BASE_URL + "/" + id);
  }

  updateRecipient(recipient, id) {
    return axios.put(RECIPIENT_API_BASE_URL + "/" + id, recipient);
  }
}

export default new RecipientService();
